src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,834 @@
+
+            var dropColumnsLength = ctx.Reader.ReadInt32();
+            if (dropColumnsLength > 0)
+            {
+                _columnsToDrop = new DateTimeEstimator.ColumnsProduced[dropColumnsLength];
+                //read in enum bytes
+                for (int i = 0; i < dropColumnsLength; i++)
+                    _columnsToDrop[i] = (DateTimeEstimator.ColumnsProduced)ctx.Reader.ReadByte();
+            }
+
+            _activeColumnMapping = new byte[Enum.GetValues(typeof(DateTimeEstimator.ColumnsProduced)).Length - dropColumnsLength];
+            var index = 0;
+            foreach (DateTimeEstimator.ColumnsProduced column in Enum.GetValues(typeof(DateTimeEstimator.ColumnsProduced)))
+            {
+                if (_columnsToDrop == null || !_columnsToDrop.Contains(column))
+                {
+                    _activeColumnMapping[index++] = (byte)column;
+                }
+            }
+
+            var dataLength = ctx.Reader.ReadInt32();
+            var data = ctx.Reader.ReadByteArray(dataLength);
+            _column.CreateTransformerFromSavedData(data);
+        }
+
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, DataViewSchema inputSchema)
+            => new DateTimeTransformer(env, ctx).MakeRowMapper(inputSchema);
+
+        private protected override IRowMapper MakeRowMapper(DataViewSchema schema) => new Mapper(this, schema);
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "DATETI T",
+                verWrittenCur: 0x00010001,
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature,
+                loaderAssemblyName: typeof(DateTimeTransformer).Assembly.FullName);
+        }
+
+        private protected override void SaveModel(ModelSaveContext ctx)
+        {
+
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // name of input column
+            // column prefix
+            // byte length of columns to drop array
+            // byte array of columns to drop
+            // length of C++ state array
+            // C++ byte state array
+
+            ctx.Writer.Write(_column.Source);
+            ctx.Writer.Write(_column.Prefix);
+
+            ctx.Writer.Write(_columnsToDrop == null ? 0 : _columnsToDrop.Length);
+            if (_columnsToDrop != null)
+            {
+                foreach (var toDrop in _columnsToDrop)
+                    ctx.Writer.Write((byte)toDrop);
+            }
+
+            var data = _column.CreateTransformerSaveData();
+            ctx.Writer.Write(data.Length);
+            ctx.Writer.Write(data);
+        }
+
+        public void Dispose()
```

Comment:
Who calls this Dispose method? We have a problem with tranformers that need to be disposed. See [https://github.com/dotnet/machinelearning/issues/906](https://github.com/dotnet/machinelearning/issues/906) and my comment on [https://github.com/dotnet/machinelearning/pull/4223](https://github.com/dotnet/machinelearning/pull/4223)

cc @codemsz

Reply:
It should just be called automatically when the pipeline teardown happens as mentioned in that comment chain. I am not manually calling it anywhere. Is it happening during pipeline teardown ok? Or is there more desirable behavior?

Reply:
> It should just be called automatically when the pipeline teardown happens as mentioned in that comment chain.

There is no call of it when the pipeline teardown happens. That is why that bug was opened. I'm still not certain on why it was closed. Maybe @codemzs  can comment.